pages/add/add.jsx

```diff
@@ -1,12 +1,201 @@
+          let redirectUrl = `${env.ORIGIN}${this.props.router.getCurrentLocation().pathname}`;
+
+          window.location.href = UserData.logInUrl.get(redirectUrl);
+        }
+      })
+      .catch(reason => {
+        this.updateLoggedInInfo(false);
+        console.error(reason);
+      });
+  },
+  updateLoggedInInfo(status, username = ``) {
+    UserData.userLoggedInStatus.set(status);
+    UserData.username.set(username);
+
+    // we have verified user's logged in status with Pulse API
+    this.setState({
+      loggedInStatusChecked: true
+    });
+
+    // After the login process user will be redirected back to Pulse.
+    // There will be a query argument "loggedin" presented in the URL
+    // which is something we don't want it to stay at all times.
+    this.clearLoggedinParam();
+  },
+  clearLoggedinParam() {
+    let query = this.props.router.location.query;
+
+    // we don't need the "loggedin" param sent back from Pulse API to be presented at all times
+    delete query.loggedin;
+
+    browserHistory.replace({
+      pathname: this.props.router.location.pathname,
+      query: query
+    });
+  },
+  logOutUser(event) {
+    event.preventDefault();
+    Service.logout()
+      .then(() => {
+        // we have successfully logged user out
+        this.updateLoggedInInfo(false);
+        // redirect user to /featured page
+        browserHistory.push({
+          pathname: `/featured`
+        });
+      })
+      .catch(reason => {
+        console.error(reason);
+        // TODO:FIXME: what do we do here?
+      });
+  },
+  handleCreatorClick(event) {
+    event.preventDefault();
+    this.setState({numCreatorFields: this.state.numCreatorFields+1});
+  },
   render() {
+    let creatorFields = [];
+
+    for (var i=0; i<this.state.numCreatorFields; i++) {
+      creatorFields.push( <CreatorInputField key={i} />);
+    }
+
+    let ContentForUnauthedUser = ( <HintMessage imgSrc={`/assets/svg/icon-bookmark-selected.svg`}
+                                               header={`Sign in failed`}
+                                               btn={{to: `/featured`, text: `Explore featured`}}>
+                                    <p>Only Mozilla staff can login now as we test this new platform. Check back soon!</p>
+                                  </HintMessage>);
+
+    let ContentForAuthedUser = (
+      <div>
+        <h1>Share with the Network</h1>
+        <p>Do you have something to share? If it might be useful to someone in our network, share it here! Pulse includes links to products and software tools, research reports and findings, think pieces, white papers, interviews, and curricula. If it might be useful, share it … at any stage or fidelity.</p>
+        <form action="" method="post">
+          <h2>Basic Info</h2>
+          <div className="posted-by">
+            <p>Posted by:&nbsp;<span className="user-full-name">{UserData.username.get()}</span></p>
```

Comment:
"Posting as" - the posting has not happened yet, so "posted by" is a little odd. Also we don't need the `&nbsp;` here, you only need that if the `<span>` is on the next line (because whitespace at the end of lines gets ignored.